solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -236,6 +240,21 @@ public void prepare(ResponseBuilder rb) throws IOException
     if (rb.getSortSpec().getOffset() < 0) {
       throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "'start' parameter cannot be negative");
     }
+    if(rb.getSortSpec().getCount() > ROWS_WARN_THRESHOLD && shouldLogPeriodically("rowsWarn", 60)) {
+      log.warn("Very high 'rows' parameter detected. This may lead to performance- and memory problems. " +
+          "Consider pagination, see https://solr.apache.org/guide/pagination-of-results.html. " +
+          "This warning will mute for 60s.");
```

Comment:
Should the default mute be 3600s instead? 24 WARN logs per day should be enough to draw attention to this issue in most systems.

Reply:
I think maybe 300s is a better compromise. I'd probably miss something that logs hourly, based on my experiences with heavily used clusters - there are plenty of WARN messages already about slow queries, leader elections, other "normal" operations in a large cluster.

Reply:
How about making the logging interval (and the threshold) configurable? I've put up a possible approach in https://github.com/cominvent/solr/compare/solr15252-log-high-rows...cpoerschke:solr15252-log-high-rows and the idea would be that both could be made default (or invariant) parameters on the request handler. Admittedly `rows.warnThresholdLogIntervalSeconds` is a bit of a long name though.